### PR TITLE
Simplify tests.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Extracted from https://github.com/Homebrew/brew/pull/16629 to see if any of these changes cause the flaky tests.

### CODE REVIEW

File: Library/Homebrew/test/support/helper/spec/shared_context/homebrew_cask.rb [hunk 1]
```diff
@@ -39,15 +39,26 @@ class Config
...
-        tap.cask_dir.mkpath
-        (TEST_FIXTURE_DIR/"cask/Casks").children.each do |casks_path|
-          FileUtils.ln_sf casks_path, tap.cask_dir
+        fixture_cask_dir = TEST_FIXTURE_DIR/"cask/Casks"
+        fixture_cask_dir.glob("**/*.rb").each do |fixture_cask_path|
+          relative_cask_path = fixture_cask_path.relative_path_from(fixture_cask_dir)
+
+          # These are only used manually in tests since they
+          # would otherwise conflict with other casks.
+          next if relative_cask_path.dirname.basename.to_s == "outdated"
```
Comment (apainintheneck) [inline Library/Homebrew/test/support/helper/spec/shared_context/homebrew_cask.rb:48]: I wonder if we could just include this in the glob pattern itself. Doesn't need to be changed though. This is fine.

**Review by apainintheneck — APPROVED** [review]
Looks good to me. At a glance, it doesn't seem to increase test time noticeably and it makes the tests easier to follow. It also better insulates tests from each other by clearing the tap caches.

**Comment by p-linnane** [comment]
This may be causing some `brew style` errors. It appears it's not respecting the sharding we did of the cask repo:
https://github.com/Homebrew/homebrew-core/actions/runs/7861902512/job/21452740091?pr=162322#step:3:35

**Comment by apainintheneck** [comment]
Interesting... I wouldn't expect this to affect the linter at all since the only things that have changed here are tests and fixtures. I'll look into it some more but also we could just revert this since it's just a refactor.

**Comment by p-linnane** [comment]
This is very much a best guess on my end for which recent PR changed this behavior. I don't think it's affecting enough stuff to revert, since that one instance is all I've seen having issues. If we can fix forward that'd be even better!

**Comment by apainintheneck** [comment]
It seems that there exists a handbrake cask as well and that's getting linted here for no reason. It'd probably be good to fix the test bot so that it only runs then audit on formulae here as well. But that doesn't really solve the root of the problem, just the symptom.

**Pull request merged**